gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -269,32 +296,55 @@ private boolean isPathWhiteListed(String path)
                 || extraWhitelistPaths.stream().anyMatch(s -> path.startsWith(s));
     }
 
-    public boolean handleAuthRequest(HttpServletRequest request)
+    @Override
+    public Optional<Cookie> removeCookie(HttpServletRequest clientRequest)
     {
-        return true;
+        if (!removeCookiePaths.contains(clientRequest.getRequestURI())) {
```

Comment:
> cookiePaths: If the request URI starts with a path in this list, add a session cookie if none exists. If a cookie exists, route the request to the backend associated with that cookie.

Docs states this is a  startsWith check, but the actual logic is doing a full match check?

Reply:
good catch, same for `setCookie` as well